phpunit/interactivity-api/class-wp-interactivity-api-directives-processor-test.php

```diff
@@ -366,4 +366,354 @@ public function test_is_void_element() {
+		$p->next_tag();
+		$result = $p->append_content_after_closing_tag_on_balanced_or_void_tags( $new_content );
+		$this->assertTrue( $result );
+		$this->assertEquals( $content . $new_content, $p );
+	}
+
+	/**
+	 * Tests that the `next_balanced_tag_closer_tag` method finds a closing tag
+	 * for a standard tag.
+	 *
+	 * @covers ::next_balanced_tag_closer_tag
+	 */
+	public function test_next_balanced_tag_closer_tag_standard_tags() {
+			$content = '<div>Text</div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertTrue( $p->next_balanced_tag_closer_tag() );
+			$this->assertEquals( 'DIV', $p->get_tag() );
+			$this->assertTrue( $p->is_tag_closer() );
+	}
+
+	/**
+	 * Tests that the `next_balanced_tag_closer_tag` method returns false for a
+	 * self-closing tag.
+	 *
+	 * @covers ::next_balanced_tag_closer_tag
+	 */
+	public function test_next_balanced_tag_closer_tag_void_tag() {
+			$content = '<img src="image.jpg" />';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertFalse( $p->next_balanced_tag_closer_tag() );
+
+			$content = '<img src="image.jpg" /><div>Text</div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertFalse( $p->next_balanced_tag_closer_tag() );
+	}
+
+	/**
+	 * Tests that the `next_balanced_tag_closer_tag` method correctly handles
+	 * nested tags.
+	 *
+	 * @covers ::next_balanced_tag_closer_tag
+	 */
+	public function test_next_balanced_tag_closer_tag_nested_tags() {
+			$content = '<div><span>Nested content</span></div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertTrue( $p->next_balanced_tag_closer_tag() );
+			$this->assertEquals( 'DIV', $p->get_tag() );
+			$this->assertTrue( $p->is_tag_closer() );
+
+			$content = '<div><div>Nested content</div></div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertTrue( $p->next_balanced_tag_closer_tag() );
+			$this->assertEquals( 'DIV', $p->get_tag() );
+			$this->assertTrue( $p->is_tag_closer() );
+			$this->assertFalse( $p->next_tag() ); // No more content.
+	}
+
+	/**
+	 * Tests that the `next_balanced_tag_closer_tag` method returns false when no
+	 * matching closing tag is found.
+	 *
+	 * @covers ::next_balanced_tag_closer_tag
+	 */
+	public function test_next_balanced_tag_closer_tag_no_matching_closing_tag() {
+			$content = '<div>No closing tag here';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+
+			$content = '<div><div>No closing tag here</div>';
+			$p       = new WP_Interactivity_API_Directives_Processor( $content );
+			$p->next_tag();
+			$this->assertFalse( $p->next_balanced_tag_closer_tag() );
+		$this->assertFalse( $p->next_balanced_tag_closer_tag() );
```

Comment:
This line probably was intended to be after 696. ☝️

Reply:
Thanks! Fixed in https://github.com/WordPress/gutenberg/pull/58498/commits/72a2f19de85622bfa4274aa5074776855c3b38c7.